src/GitTools.Core/Context/AuthenticationContext.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace GitTools
+{
+    public class AuthenticationContext : IAuthenticationContext
+    {
+        public AuthenticationContext()
+        {
+        }
+
+        public AuthenticationContext(string username, string password)
+        {
+            Username = username;
+            Password = password;
+        }
+
+        public AuthenticationContext(string token)
+        {
+            Token = token;
+        }
+
+        public string Username { get; set; }
+        public string Password { get; set; }
```

Comment:
Should this be a SecureString? https://msdn.microsoft.com/en-us/library/system.security.securestring(v=vs.110).aspx